src/main/resources/openapi.json

```diff
@@ -1,286 +1,935 @@
+                    }
+                }
+            ]
         }
-      ]
     },
-    "/consumers/{groupid}/instances/{name}/partitions/end": {
-      "post": {
-        "operationId": "seekToEnd",
-        "responses": {
-          "200": {
-            "description": "OK"
-          }
+    "components": {
+        "schemas": {
+            "ApiResponse": {
+                "type": "object",
+                "properties": {
+                    "code": {
+                        "format": "int32",
+                        "type": "integer"
+                    },
+                    "type": {
+                        "type": "string"
+                    },
+                    "message": {
+                        "type": "string"
+                    }
+                }
+            },
+            "Consumer": {
+                "title": "Root Type for Consumer",
+                "description": "The root of the Consumer type's schema.",
+                "type": "object",
+                "properties": {
+                    "name": {
+                        "type": "string"
+                    },
+                    "format": {
+                        "type": "string"
+                    },
+                    "auto.offset.reset": {
+                        "type": "string"
+                    },
+                    "auto.commit.enable": {
+                        "type": "string"
+                    },
+                    "fetch.min.bytes": {
+                        "type": "string"
+                    },
+                    "consumer.request.timeout.ms": {
+                        "type": "string"
+                    }
+                },
+                "example": "{\n    \"name\": \"consumer1\",\n    \"format\": \"binary\",\n    \"auto.offset.reset\": \"earliest\",\n    \"auto.commit.enable\": \"false\",\n    \"fetch.min.bytes\": \"512\",\n    \"consumer.request.timeout.ms\": \"30000\"\n}"
+            },
+            "Offset": {
+                "title": "Offset",
+                "required": [],
+                "type": "object",
+                "properties": {
+                    "partition": {
+                        "type": "integer"
+                    },
+                    "offset": {
+                        "type": "integer"
+                    },
+                    "topic": {
+                        "type": "string"
+                    }
+                }
+            },
+            "Partition": {
+                "title": "Partition",
+                "type": "object",
+                "properties": {
+                    "partition": {
+                        "format": "int32",
+                        "type": "integer"
+                    },
+                    "leader": {
+                        "format": "int32",
+                        "type": "integer"
+                    },
+                    "replicas": {
+                        "type": "object"
```

Comment:
Can this be constrained further than "object"? And is it really an object or an array of objects?

Reply:
I don't think we need this custom type at all, it's about a GET operations on topics endpoint which we don't support.

Reply:
It is used as data type of array in `Partitions`.

Reply:
And we don't need `Partitions` as well